File: 2661-smallest-missing-non-negative-integer-after-operations/smallest-missing-non-negative-integer-after-operations.go
func findSmallestInteger(nums []int, value int) int {
	counter := make(map[int]int)
	for i, _ := range nums {
		nums[i] %= value
		if nums[i] < 0 {
			nums[i] += value
		}
		counter[nums[i]] += 1
	}
    
	for i := 0; i <= len(nums); i++ {
		if c, ok := counter[i%value]; !ok || c <= 0 {
			return i
		} else {
			counter[i%value] -= 1
		}
	}

	return 0
}

File: 2661-smallest-missing-non-negative-integer-after-operations/smallest-missing-non-negative-integer-after-operations.py
class Solution:
    def findSmallestInteger(self, nums: List[int], value: int) -> int:
        counter = Counter([x % value for x in nums])
        ans = 0
        while counter[ans % value] > 0:
            counter[ans % value] -= 1
            ans += 1

        return ans